gap/WreathProductElements.gi

```diff
@@ -31,8 +31,47 @@ function(g, G)
     return ForAll([1 .. Length(l) - 1], i -> l[i] in K) and l[Length(l)] in H;
 end);
 
-# InstallMethod( \in, "perm, and perm wreath product", true,[IsObject, IsPermGroup and HasWreathProductInfo], OVERRIDENICE + 42, WPE_IN);
-# InstallMethod( \in, "matrix, and matrix wreath product", true, [IsMatrix, IsMatrixGroup and HasWreathProductInfo], OVERRIDENICE + 42, WPE_IN);
+InstallMethod( \in, "perm, and perm wreath product", true,[IsPerm, IsPermGroup and HasWreathProductInfo], OVERRIDENICE + 42,
+function(g, G)
+    local info, comps, n, i;
+    info := WreathProductInfo(G);
+    comps := ComponentsOfWreathProduct(G);
+    n := NrMovedPoints(comps[2]);
+    if not IsBound(info.projection) or Length(info.embeddings) < n + 1 then
+        if not IsBound(info.constructingMaps) then
+            info.constructingMaps := true;
+            for i in [1 .. n + 1] do
+                Embedding(G, i);
+            od;
+            Projection(G);
+            info.constructingMaps := false;
+        elif info.constructingMaps then
+            TryNextMethod();
+        fi;
+    fi;
```

Comment:
To avoid the code duplication, just move this code into `WPE_IN`. Yes, the `TryNextMethod()` will be fine: there is no fundamental difference between
```
InstallMethod(bla, ..., function(x) ... end);
```
and
```
foo := function(x) ... end;
InstallMethod(bla, ..., foo);
```

Reply:
Ah, I didn't know. Thanks!